src/psyclone/tests/psyir/nodes/reference_test.py

```diff
@@ -194,3 +195,160 @@ def test_array_create_invalid3():
     assert (
         "child of children argument in create method of Array class "
         "should be a PSyIR Node but found 'str'." in str(excinfo.value))
+
+
+def test_array_is_full_range():
+    '''Test that the is_full_range method in the Array Node works as
+    expected. '''
+    zero = Literal("0", DataType.INTEGER)
+    one = Literal("1", DataType.INTEGER)
+    symbol = DataSymbol("my_array", DataType.REAL, shape=[10])
+    reference = Reference(symbol)
+    lbound = BinaryOperation.create(BinaryOperation.Operator.LBOUND,
+                                    reference, one)
+    ubound = BinaryOperation.create(BinaryOperation.Operator.UBOUND,
+                                    reference, one)
+    symbol_error = DataSymbol("another_array", DataType.REAL, shape=[10])
+    reference_error = Reference(symbol_error)
+
+    # Index out of bounds
+    array_reference = Array.create(symbol,[one])
+    with pytest.raises(ValueError) as excinfo:
+        array_reference.is_full_range(1)
+    assert ("In Array 'my_array' the specified index '1' must be less than "
+            "the number of dimensions '1'." in str(excinfo.value))
+
+    # Array dimension is not a Range
+    assert not array_reference.is_full_range(0)
+
+    # Check LBOUND
+    # Array dimension range lower bound is not a binary operation
+    my_range = Range.create(one, one, one)
+    array_reference = Array.create(symbol,[my_range])
+    assert not array_reference.is_full_range(0)
+
+    # Array dimension range lower bound is not an LBOUND binary operation
+    my_range = Range.create(ubound, one, one)
+    array_reference = Array.create(symbol,[my_range])
+    assert not array_reference.is_full_range(0)
+
+    # Array dimension range lower bound is an LBOUND binary operation
+    # with the first value not being a reference
+    lbound_error = BinaryOperation.create(BinaryOperation.Operator.LBOUND,
```

Comment:
We shouldn't be able to create such an operation. Will @sergisisio's latest PR prevent this?

Reply:
No I think Sergi's PR will look at the children of BinaryOperation and won't distinguish one operator from another. Do you have a suggestion to change this?